### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by djtimmy679**
I think I added most, if not all, the remaining wrapper containers in containers.ts. There are also some configuration edits for changing the unit tests from JS to TS, but am still working on finishing the unit tests.

### CODE REVIEW

File: test/defineInput.spec.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import {defineInput} from '../src/utils/defineInput'
```
Comment (michaeltintiuc) [inline test/defineInput.spec.ts:1]: This file needs cleanup, perhaps add the test dir to be included in `npm run lint` too

File: package.json [hunk 2]
```diff
@@ -66,10 +69,15 @@
...
     "rollup-plugin-typescript2": "^0.27.0",
+    "terser-webpack-plugin": "^3.0.2",
+    "ts-jest": "^26.1.0",
+    "ts-loader": "^7.0.5",
     "tslib": "^1.11.2",
     "tslint": "^6.1.2",
     "tslint-ionic-rules": "^0.0.21",
-    "typescript": "^3.8.3"
+    "typescript": "^3.9.3",
+    "webpack": "^4.43.0",
```
Comment (michaeltintiuc) [inline package.json:79]: Not sure why we need webpack, our builds are done using rollup

File: package.json [hunk 3]
```diff
@@ -53,10 +53,13 @@
     "prod": "npm run lint && npm run clean && NODE_ENV=production rollup -c ./build/rollup.config.js --configProd",
     "lint": "tslint --project . --fix",
     "test": "jest --coverage --verbose",
-    "clean": "node ./scripts/clean.js"
+    "clean": "node ./scripts/clean.js",
+    "build": "webpack"
```
Comment (michaeltintiuc) [inline package.json:57]: builds are done with rollup via commands `dev` and `prod`

File: test/ionic-vue.spec.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { IonicVue  } from '../src/ionic-vue';
```
Comment (michaeltintiuc) [inline test/ionic-vue.spec.ts:1]: Needs cleanup with lint

File: tsconfig.json [hunk 5]
```diff
@@ -2,11 +2,13 @@
   "compilerOptions": {
     "alwaysStrict": true,
     "strict": true,
+    "allowJs": true,
```
Comment (michaeltintiuc) [inline tsconfig.json:5]: Why allow js?

File: types/components/back-button/index.d.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { FunctionalComponent } from 'vue';
```
Comment (michaeltintiuc) [inline types/components/back-button/index.d.ts:1]: All type files should be removed from github, there's no point in having them here

File: tsconfig.json (hunk 5 above)
Comment (djtimmy679) [reply tsconfig.json:5]: hm, not sure where that came from

File: jest.config.js [hunk 7]
```diff
@@ -1,12 +1,11 @@
 module.exports = {
```
Comment (michaeltintiuc) [inline jest.config.js:1]: Please cleanup this file, ie remove empty lines at the end

File: package.json [hunk 8]
```diff
@@ -66,13 +68,18 @@
     "rollup-plugin-buble": "^0.19.2",
     "rollup-plugin-terser": "^1.0.1",
     "rollup-plugin-typescript2": "^0.27.0",
+    "terser-webpack-plugin": "^3.0.2",
```
Comment (michaeltintiuc) [inline package.json:71]: Remove this webpack plugin

File: test/defineContainer.spec.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import { defineContainer } from '../src/utils/defineContainer';
```
Comment (michaeltintiuc) [inline test/defineContainer.spec.ts:1]: Looks like test files aren't being linted, I see 4 char indents and missing semicolons

File: test/ionic-vue.spec.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import { IonicVue } from '../src/ionic-vue';
```
Comment (michaeltintiuc) [inline test/ionic-vue.spec.ts:1]: Update with latest `dev` and fix this test, there was a bugfix merged

File: tsconfig.json [hunk 11]
```diff
@@ -21,15 +22,16 @@
     "noImplicitReturns": true,
     "noUnusedLocals": true,
     "noUnusedParameters": true,
-    "outDir": "dist",
+    "outDir": "./dist/",
```
Comment (michaeltintiuc) [inline tsconfig.json:25]: No point in changing this

File: tsconfig.json [hunk 12]
```diff
@@ -21,15 +22,16 @@
...
     "pretty": true,
     "removeComments": false,
     "rootDir": "src",
+    "sourceMap": true,
```
Comment (michaeltintiuc) [inline tsconfig.json:29]: Why was this added?

File: tsconfig.json [hunk 13]
```diff
@@ -7,6 +7,7 @@
     "declaration": true,
     "declarationMap": true,
     "declarationDir": "types",
+    "esModuleInterop": true,
```
Comment (michaeltintiuc) [inline tsconfig.json:10]: Why was this added?

File: tsconfig.json [hunk 14]
```diff
@@ -21,15 +22,16 @@
...
     "pretty": true,
     "removeComments": false,
     "rootDir": "src",
+    "sourceMap": true,
     "strictPropertyInitialization": false,
     "target": "es5"
   },
   "include": [
-    "src/**/*.ts"
+    "src/**/*"
```
Comment (michaeltintiuc) [inline tsconfig.json:34]: Why change this?

File: tslint.json [hunk 15]
```diff
@@ -1,7 +1,8 @@
...
-  "rules": {
-    "no-non-null-assertion": false,
-    "no-import-side-effect": [true, {"ignore-module": "(svg|\\.css)$"}]
+    "extends": "tslint-ionic-rules",
+    "rules": {
+      "no-non-null-assertion": false,
+      "no-import-side-effect": [true, {"ignore-module": "(svg|\\.css)$"}]
+    }
   }
-}
```
Comment (michaeltintiuc) [inline tslint.json:7]: Looks like broken json to me

File: webpack.config.js [hunk 16]
```diff
@@ -0,0 +1,29 @@
+const path = require('path');
```
Comment (michaeltintiuc) [inline webpack.config.js:1]: remove webpack config as we're not using it

File: tsconfig.json (hunk 13 above)
Comment (djtimmy679) [reply tsconfig.json:10]: i think I ran into some compability issue and this fixed it. Not sure if we actually need it though
